### PR TITLE
Simple trim and singleton fixes.

### PR DESCRIPTION
**Pull request by sorensenjs**

### CODE REVIEW

File: evaluation/fix_spans_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    def test_ignore_space_in_span(self):
+        self.assertEqual(fix_spans._fix_spans(
+            [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14],
+            'To infinity and   beyond.'),
+                         [3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14])
+
+    def test_trailing_spaces(self):
+        self.assertEqual(fix_spans._fix_spans(
+            [3, 4, 5, 6, 7, 8, 9, 10, 12, 13, 14, 15, 16, 17, 19, 20, 21, 22,
```
Comment (LeoLaugier) [inline evaluation/fix_spans_test.py:25]: Here the third span [19, 20, 21, 22, 23, 24] "eyond." missed the b and does not add it back.

File: evaluation/fix_spans.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    cleaned = []
+    for begin, end in _contiguous_ranges(spans):
+        while text[begin]==' ' and begin < end:
+            begin += 1
+        while text[end]==' ' and begin < end:
+            end -= 1
+        if end - begin > 1:
+            cleaned.extend(range(begin, end + 1))
+    return cleaned
+
```
Comment (LeoLaugier) [inline evaluation/fix_spans.py:33]:
```suggestion


```

File: evaluation/fix_spans_test.py (hunk 1 above)
Comment (sorensenjs) [reply evaluation/fix_spans_test.py:25]: While that was my mistake in choosing the offset, I am not currently moving boundaries within words, only dealing with white space. We could do that but that is a different discussion I think.

Comment (LeoLaugier) [reply evaluation/fix_spans_test.py:25]: It's clear, thanks!

File: evaluation/fix_spans.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Lint as: python3
+"""Automatic best effort span cleaner for SemEval 2021 Toxic Spans."""
+
+import ast
+import csv
+import itertools
+import sys
+
+
```
Comment (LeoLaugier) [inline evaluation/fix_spans.py:11]:
```suggestion
SPECIAL_CHARACTERS = [" ", "\n", "\t"]

```

File: evaluation/fix_spans.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    """Extracts continguous runs [1, 2, 3, 5, 6, 7] -> [(1,3), (5,7)]."""
+    output = []
+    for _, span in itertools.groupby(
+        enumerate(span_list), lambda p: p[1] - p[0]):
+        span = list(span)
+        output.append((span[0][1], span[-1][1]))
+    return output
+
+
+def _fix_spans(spans, text):
```
Comment (LeoLaugier) [inline evaluation/fix_spans.py]:
```suggestion
def _fix_spans(spans, text, special_characters=SPECIAL_CHARACTERS):
```

File: evaluation/fix_spans.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    """Applies minor edits to trim spans and remove singletons."""
+    cleaned = []
+    for begin, end in _contiguous_ranges(spans):
+        while text[begin]==' ' and begin < end:
```
Comment (LeoLaugier) [inline evaluation/fix_spans.py]:
```suggestion
        while text[begin] in special_characters and begin < end:
```

File: evaluation/fix_spans.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            begin += 1
+        while text[end]==' ' and begin < end:
```
Comment (LeoLaugier) [inline evaluation/fix_spans.py]:
```suggestion
        while text[end] in special_characters and begin < end:
```

File: evaluation/fix_spans_test.py [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+             23, 24], 'To infinity and   beyond.'),
+                         [3, 4, 5, 6, 7, 8, 9, 10, 12, 13, 14, 19, 20, 21, 22,
+                          23, 24])
+
+    def test_drop_singletons(self):
+        self.assertEqual(fix_spans._fix_spans(
+            [3, 4, 5, 6, 7, 8, 9, 10, 13, 19, 20, 21, 22, 23, 24],
+            'To infinity and    beyond.'),
+                         [3, 4, 5, 6, 7, 8, 9, 10, 19, 20, 21, 22, 23, 24])
+
```
Comment (LeoLaugier) [inline evaluation/fix_spans_test.py:41]:
```suggestion
    def test_special_characters(self):
        self.assertEqual(fix_spans._fix_spans(
            [3, 4, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15, 16, 17, 18, 19, 21, 22,
             23, 24, 25, 26], 'To \tinfinity and\n   beyond.'),
                         [4, 5, 6, 7, 8, 9, 10, 11, 13, 14, 15, 21, 22,
                          23, 24, 25, 26])

```

Comment (LeoLaugier) [reply evaluation/fix_spans_test.py:41]: I suggest a robust resistance to special characters in addition to spaces, but it might be overkill.

**Review by LeoLaugier — APPROVED** [review]
I suggest a robust resistance to special characters in addition to spaces, but it might be overkill.

File: evaluation/fix_spans.py (hunk 3 above)
Comment (ipavlopoulos) [reply evaluation/fix_spans.py:11]: This could switch to {" ", "\n", "\t"} for a more efficient search (e.g., line 26).

**Pull request merged**